lib/pages/home_page.dart

```diff
@@ -16,11 +20,43 @@ class HomePage extends StatefulWidget {
 
 class _HomePageState extends State<HomePage> {
   Position? pos;
+  Marker daily = Marker(point: LatLng(0, 0), builder: (context) => Container());
+  int distance =
+      DistCalculator.instance.getDist(Location(0, 0), Location(0, 0));
+
+  Timer? _timer;
 
   Future<void> getPosition() async {
     pos = await Geolocation.instance.position;
   }
 
+  Future<void> initiateChallenge() async {
+    setState(() {
+      daily = DistCalculator.instance
+          .initiateChallenge(2500, 500000, LatLng(pos!.latitude, pos!.longitude));
```

Comment:
Bylo by fajn mít takové hodnoty (2500, 500000) někdě uložené jako konstanty. Bude to pak mnohem lépe ovladatelnější, dohledatelnější a také čitelnější (díky jménu proměnné)